pkg/cmd/start/start.go

```diff
@@ -35,59 +42,126 @@ func NewOperator() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "start",
 		Short: "Start the operator",
-		Run: func(cmd *cobra.Command, args []string) {
-			// boiler plate for the "normal" command
-			rand.Seed(time.Now().UTC().UnixNano())
-			defer serviceability.BehaviorOnPanic(os.Getenv("OPENSHIFT_ON_PANIC"), version.Get())()
-			defer serviceability.Profile(os.Getenv("OPENSHIFT_PROFILE")).Stop()
-			serviceability.StartProfiler()
-
-			if config := cmd.Flags().Lookup("config").Value.String(); len(config) == 0 {
-				klog.Fatalf("error: --config is required")
-			}
-
-			unstructured, config, configBytes, err := cfg.Config()
-			if err != nil {
-				klog.Fatal(err)
-			}
+		Run:   runOperator(operator, cfg),
+	}
+	cmd.Flags().AddFlagSet(cfg.NewCommand().Flags())
 
-			startingFileContent, observedFiles, err := cfg.AddDefaultRotationToConfig(config, configBytes)
-			if err != nil {
-				klog.Fatal(err)
-			}
+	return cmd
+}
 
-			// if the service CA is rotated, we want to restart
-			if data, err := ioutil.ReadFile(serviceCACertPath); err == nil {
-				startingFileContent[serviceCACertPath] = data
-			} else {
-				klog.V(4).Infof("Unable to read service ca bundle: %v", err)
-			}
-			observedFiles = append(observedFiles, serviceCACertPath)
-
-			exitOnChangeReactorCh := make(chan struct{})
-			ctx := context.Background()
-			ctx2, cancel := context.WithCancel(ctx)
-			go func() {
-				select {
-				case <-exitOnChangeReactorCh:
-					cancel()
-				case <-ctx.Done():
-					cancel()
-				}
-			}()
-
-			builder := controllercmd.NewController("openshift-insights-operator", operator.Run).
-				WithKubeConfigFile(cmd.Flags().Lookup("kubeconfig").Value.String(), nil).
-				WithLeaderElection(config.LeaderElection, "", "openshift-insights-operator-lock").
-				WithServer(config.ServingInfo, config.Authentication, config.Authorization).
-				WithRestartOnChange(exitOnChangeReactorCh, startingFileContent, observedFiles...)
-
-			if err := builder.Run(ctx2, unstructured); err != nil {
-				klog.Fatal(err)
-			}
+func NewGather() *cobra.Command {
+	operator := &controller.GatherJob{
+		Controller: config.Controller{
+			StoragePath: "/var/lib/insights-operator",
+			Interval:    10 * time.Minute,
+			Endpoint:    "local",
```

Comment:
Can we just omit this?

Reply:
It used to be necessary because if it wasn't set then we would be 'Disabled', but no longer necessary